pkg/login/social/connectors/azuread_oauth.go

```diff
@@ -163,6 +163,16 @@ func (s *SocialAzureAD) UserInfo(ctx context.Context, client *http.Client, token
 }
 
 func (s *SocialAzureAD) Validate(ctx context.Context, settings ssoModels.SSOSettings) error {
+	info, err := CreateOAuthInfoFromKeyValues(settings.Settings)
+	if err != nil {
+		return ssosettings.ErrInvalidSettings.Errorf("SSO settings map cannot be converted to OAuthInfo: %v", err)
+	}
+
+	err = validateInfo(info)
+	if err != nil {
+		return err
+	}
+
 	return nil
```

Comment:
```suggestion
	return validateInfo(info)
```

nit

Reply:
applies to the other connectors as well, no need to fix

Reply:
I did this because there will be specific validations in the future for each connector, like so
```
// this is the common validation for all providers
err = validateInfo(info)
if err != nil {
	return err
}

// specific provider validation rules
if info.UsePKCE == false {
  return errors.New("PKCE must be set")
}
return nil
```